aten/src/ATen/core/jit_type.h

```diff
@@ -1229,6 +1230,23 @@ struct TORCH_API ComplexType : public NumberType {
   }
 };
 
+struct SymbolicOrConcreteIntType;
+using SymbolicOrConcreteIntTypePtr = SingletonTypePtr<SymbolicOrConcreteIntType>;
+struct TORCH_API SymbolicOrConcreteIntType : public Type {
+  bool equals(const Type& rhs) const override {
+    return rhs.kind() == kind();
+  }
+  std::string str() const override {
+    return "BoxedInt";
+  }
+  static const TypeKind Kind = TypeKind::SymbolicOrConcreteIntType;
+  // global singleton
+  static SymbolicOrConcreteIntTypePtr get();
+
+ private:
+  SymbolicOrConcreteIntType() : Type(TypeKind::SymbolicOrConcreteIntType) {}
+};
```

Comment:
We need to setup the subtyping relation in TorchScript so that anywhere symint is expected, an int can be provided

Reply:
narrow_copy is too niche and there probably isn't any test coverage for it in TorchScript

Reply:
I added some tests